gravitee-apim-console-webui/src/management/api/runtime-logs-v4/runtime-logs/components/api-runtime-logs-quick-filters/components/api-runtime-logs-more-filters/components/api-runtime-logs-more-filters-form.component.ts

```diff
@@ -0,0 +1,94 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ChangeDetectorRef, Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { takeUntil, tap } from 'rxjs/operators';
+import { Moment } from 'moment';
+
+import { DEFAULT_PERIOD, MoreFiltersForm, PERIODS } from '../../../../../models';
+
+@Component({
+  selector: 'api-runtime-logs-more-filters-form',
+  template: require('./api-runtime-logs-more-filters-form.component.html'),
+  styles: [require('./api-runtime-logs-more-filters-form.component.scss')],
+})
+export class ApiRuntimeLogsMoreFiltersFormComponent implements OnInit, OnDestroy {
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+
+  @Input() formValues: MoreFiltersForm;
+  @Output() valuesChangeEvent: EventEmitter<MoreFiltersForm> = new EventEmitter<MoreFiltersForm>();
+  @Output() isInvalidEvent: EventEmitter<boolean> = new EventEmitter<boolean>();
+  readonly periods = PERIODS;
+  moreFiltersForm: FormGroup;
+  minDate: Moment;
+
+  constructor(private readonly cdr: ChangeDetectorRef) {}
+
+  ngOnInit() {
+    this.moreFiltersForm = new FormGroup({
+      period: new FormControl(this.formValues.period),
+      from: new FormControl(this.formValues.from),
+      to: new FormControl(this.formValues.to),
+    });
+    this.minDate = this.formValues.from;
+    this.onDatesChange();
+  }
+
+  ngOnDestroy(): void {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.unsubscribe();
+  }
+
+  onDatesChange() {
+    this.moreFiltersForm
+      .get('period')
+      .valueChanges.pipe(
+        tap(() => {
+          this.moreFiltersForm.get('from').setValue(null, { emitEvent: false, onlySelf: true });
+          this.moreFiltersForm.get('to').setValue(null, { emitEvent: false, onlySelf: true });
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => this.emitValues());
+
+    this.moreFiltersForm
+      .get('from')
+      .valueChanges.pipe(
+        tap((from) => {
+          this.minDate = from;
+          this.moreFiltersForm.get('period').setValue(DEFAULT_PERIOD, { emitEvent: false, onlySelf: true });
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => this.emitValues());
+
+    this.moreFiltersForm
+      .get('to')
+      .valueChanges.pipe(
+        tap(() => this.moreFiltersForm.get('period').setValue(DEFAULT_PERIOD, { emitEvent: false, onlySelf: true })),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => this.emitValues());
+  }
+
+  private emitValues() {
+    this.moreFiltersForm.updateValueAndValidity({ emitEvent: false });
+    this.isInvalidEvent.emit(this.moreFiltersForm.invalid);
+    this.valuesChangeEvent.emit(this.moreFiltersForm.getRawValue());
+    this.cdr.detectChanges();
```

Comment:
Here I added it only because of jest. In the browser everything is ok but without that we have a test that fails saying `expression has changed...`